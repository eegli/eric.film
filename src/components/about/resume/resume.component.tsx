import { useState } from 'react';
import { FaAngleDown, FaAngleUp } from 'react-icons/fa';
import SmoothCollapse from 'react-smooth-collapse';
import {
  ResumeContent,
  ResumeWrapper,
  Section,
  ToggleContainer,
  ToggleTitle,
} from './resume.styles';

const ResumeComponent: React.FC = () => {
  const [workEdOpen, setWorkEdOpen] = useState<boolean>(false);
  const [toolsTechOpen, setToolsTechOpen] = useState<boolean>(false);

  return (
    <ResumeWrapper>
      <Section>
        <ToggleContainer onClick={() => setWorkEdOpen(!workEdOpen)}>
          <ToggleTitle>education - work</ToggleTitle>
          {workEdOpen ? <FaAngleUp /> : <FaAngleDown />}
        </ToggleContainer>
        <SmoothCollapse expanded={workEdOpen}>
          <ResumeContent>
            <span>EDUCATION</span>
            <div>
              <span>2020 - 2024</span>
              <p>Master of Science, Data Science. University of Zurich</p>
              <span>2015 - 2019</span>
              <p>
                Bachelor of Arts, Media and Communication Science. University of
                Zurich
              </p>
            </div>
            <span>WORK</span>
            <div>
              <span>05.2018 - present</span>
              <p>Digital Project Manager. Ringier AG</p>
              <span>01.2013 - present</span>
              <p>Freelance photographer & videographer. Self-employed</p>
              <span>01.2017 - 04.2018</span>
              <p>Project Management Support. Ringier AG</p>
              <span>05.2016 - 04.2019</span>
              <p>Video Journalist. Zürcher Oberländer Medien AG</p>
            </div>
          </ResumeContent>
        </SmoothCollapse>
      </Section>
      <Section>
        <ToggleContainer onClick={() => setToolsTechOpen(!toolsTechOpen)}>
          <ToggleTitle>tools - technology</ToggleTitle>
          {toolsTechOpen ? <FaAngleUp /> : <FaAngleDown />}
        </ToggleContainer>
        <SmoothCollapse expanded={toolsTechOpen}>
          <ResumeContent>
            <span>DEVELOPMENT</span>
            <div>
              <ul>
                <li>Javascript (ES5, ES6)</li>
                <li>Typescript</li>
                <li>Next.js</li>
                <li>React.js, React native</li>
                <li>Redux, React Redux</li>
                <li>CSS in JS</li>
                <li>HTML5, CSS3</li>
                <li>GraphQL, REST</li>
                <li>Google Firebase</li>
                <li>Git</li>
                <li>Node.js (currently learning)</li>
              </ul>
            </div>
            <span>MISC</span>
            <div>
              <ul>
                <li>Agile (Scrum & Kanban)</li>
                <li>Adobe Creative Suite</li>
              </ul>
            </div>
          </ResumeContent>
        </SmoothCollapse>
      </Section>
    </ResumeWrapper>
  );
};

export default ResumeComponent;
