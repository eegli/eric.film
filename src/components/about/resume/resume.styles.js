import styled from 'styled-components';

const mobileBp = '1000px';

export const Container = styled.div`
  margin-top: 1rem;
  width: 93%;
  color: ${({ theme }) => theme.colors.baseBg};
  padding: 1rem;
  display: flex;
  @media screen and (max-width: ${mobileBp}) {
    flex-direction: column;
  }
`;

export const Section = styled.div`
  flex: 1;
  margin-right: 2rem;
  &:nth-child(2) {
    margin-left: 1rem;
    margin-right: 0;
    @media screen and (max-width: ${mobileBp}) {
      margin: 2rem 0 0 0;
    }
  }
  @media screen and (max-width: ${mobileBp}) {
    margin: 0;
  }
`;

export const ResumeContent = styled.div`
  background-color: ${({ theme }) => theme.colors.baseBgraised};
  color: ${({ theme }) => theme.colors.textPrimaryColor};
  padding: 1rem 2rem 1rem 1rem;
  & > span {
    color: ${({ theme }) => theme.colors.textSecondaryColor};
    font-size: ${({ theme }) => theme.fontsizes.secondaryFontSize};
  }
  & > div {
    margin-top: 0.5rem;
    padding: 0.6rem;
    & > span {
      color: ${({ theme }) => theme.colors.textSecondaryColor};
    }
    & > p {
      margin: 0 0 0.8rem 0;
    }
    & > ul {
      margin: 0 0 0.8rem 0;
      padding-left: 1.2rem;
    }
    & * li {
      margin-bottom: 0.5rem;
    }
  }
`;

export const ToggleContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: ${({ theme }) => theme.colors.textSecondaryColor};
  padding: 1rem;
  &:hover {
    cursor: pointer;
  }
`;

export const ToggleTitle = styled.p`
  display: inline;
  font-size: ${({ theme }) => theme.fontsizes.h2FontSize};
  margin: 0;
  font-weight: 600;
`;
