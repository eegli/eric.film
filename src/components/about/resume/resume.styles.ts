import styled from 'styled-components';

const mobileBp = '1000px';

export const Container = styled.div`
  margin-top: 0.5rem;
  width: 93%;
  color: ${({ theme }) => theme.colors.bgPrimary};
  padding: 1rem;
  display: flex;
  @media screen and (max-width: ${mobileBp}) {
    flex-direction: column;
  }
`;

export const Section = styled.div`
  flex: 1;
  margin-right: 1rem;
  &:nth-child(2) {
    margin-left: 1rem;
    margin-right: 0;
    @media screen and (max-width: ${mobileBp}) {
      margin: 2rem 0 0 0;
    }
  }
  @media screen and (max-width: ${mobileBp}) {
    margin: 0;
  }
`;

export const ResumeContent = styled.div`
  background-color: ${({ theme }) => theme.colors.bgSecondary};
  color: ${({ theme }) => theme.colors.textPrimary};
  padding: 1rem 2rem 1rem 1rem;
  & span {
    color: ${({ theme }) => theme.colors.textSecondary};
    font-size: ${({ theme }) => theme.fontsizes.secondaryFontSize};
  }
  & > div {
    margin-top: 0.5rem;
    padding: 0.6rem;

    & > p {
      margin: 0 0 0.8rem 0;
    }
    & > ul {
      margin: 0 0 0.8rem 0;
      padding-left: 1.2rem;
    }
    & * li {
      margin-bottom: 0.5rem;
    }
  }
`;

export const ToggleContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: ${({ theme }) => theme.colors.bgTertiary};
  color: ${({ theme }) => theme.colors.textSecondary};
  padding: 1rem;
  &:hover {
    cursor: pointer;
  }
`;

export const ToggleTitle = styled.h4`
  display: inline;
  margin: 0;
  font-weight: 400;
`;
