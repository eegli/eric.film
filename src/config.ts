const base = 'https://beta.eric.film/static/img';

// Type inference is pretty strong here
export const SEO_INDEX_META = {
  title: 'Eric Egli - Freelance Photographer & Filmer',
  description:
    'Freelance photography & filmmaking. Based in Zurich, Switzerland. Blogging about web development, video editing and more.',
  og_title: 'Eric Egli - Freelance Photographer & Filmer',
  og_image: `${base}/og/og_home.jpg`,
  og_site_name: 'Eric Egli',
  og_description:
    'Freelance photography & filmmaking. Based in Zurich, Switzerland. Blogging about web development, video editing and more.',
  twitter_card: 'summary_large_image',
  twitter_title: 'Eric Egli - Freelance Photographer & Filmer',
  twitter_description:
    'Freelance photography & filmmaking. Based in Zurich, Switzerland. Blogging about web development, video editing and more.',
  twitter_image: `${base}/og/og_home.jpg`,
};
export const SEO_ABOUT_META = {
  title: 'Eric Egli - About | Photograper & Filmmaker',
  description:
    "I'm a freelance photographer & filmer with a passion for web development. Say Hello to the team behind eric.film.",
  og_title: 'Eric Egli - About | Photograper & Filmmaker',
  og_image: `${base}/og/og_about.jpg`,
  og_site_name: 'Eric Egli',
  og_description:
    "I'm a freelance photographer & filmer with a passion for web development. Say Hello to the team behind eric.film.",
  twitter_card: 'summary_large_image',
  twitter_title: 'Eric Egli - About | Photograper & Filmmaker',
  twitter_description:
    "I'm a freelance photographer & filmer with a passion for web development. Say Hello to the team behind eric.film.",
  twitter_image: `${base}/og/og_about.jpg`,
};
export const SEO_PORTFOLIO_META = {
  title: 'Eric Egli - Portfolio | Photograper & Filmmaker',
  description:
    'Moving and still portfolio - A selection of my work in the past years. Shots from all over the world.',
  og_title: 'Eric Egli - Portfolio | Photograper & Filmmaker',
  og_image: `${base}/og/og_portfolio.jpg`,
  og_site_name: 'Eric Egli',
  og_description:
    'Moving and still portfolio - A selection of my work in the past years. Shots from all over the world.',
  twitter_card: 'summary_large_image',
  twitter_title: 'Eric Egli - Portfolio | Photograper & Filmmaker',
  twitter_description:
    'Moving and still portfolio - A selection of my work in the past years. Shots from all over the world.',
  twitter_image: `${base}/og/og_portfolio.jpg`,
};

export const SEO_BLOG_INDEX_META = {
  title: 'Eric Egli - Blog | Latest Blog Posts',
  description:
    'Check out the latest video of the week or get insights into web development with React.js, Next.js and more!',
  og_title: 'Eric Egli - Blog | Latest Blog Posts',
  og_image: `${base}/og/og_blog.jpg`,
  og_site_name: 'Eric Egli',
  og_description:
    'Check out the latest video of the week or get insights into web development with React.js, Next.js and more!',
  twitter_card: 'summary_large_image',
  twitter_title: 'Eric Egli - Blog | Latest Blog Posts',
  twitter_description:
    'Check out the latest video of the week or get insights into web development with React.js, Next.js and more!',
  twitter_image: `${base}/og/og_blog.jpg`,
};

export const SCHEMA_PUBLISHER_LOGO = {
  url: `${base}/var/schema_logo.png`,
};

export const FALLBACK_IMG = {
  url: `${base}/placeholder.jpg`,
};
