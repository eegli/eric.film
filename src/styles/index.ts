export { GlobalTheme } from './theme/global';
export type { DarkTheme as dark, LightTheme as light } from './theme/theme';
export type StyleObject<T> = { [key: string]: T };
