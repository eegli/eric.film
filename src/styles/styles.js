import { createGlobalStyle } from 'styled-components';

export const Global = createGlobalStyle`

  html {
    overflow-y: scroll;
    overflow-x: hidden;
  }

  /* GLOBALS */
  body {
    font-family: 'Roboto', sans-serif;
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    background-color: ${({ theme }) => theme.baseBg};
    color: ${({ theme }) => theme.textColor};
    height: 100vh;
    width: 100vw;
    min-width: 320px;
  }

  p {
    font-size: ${({ theme }) => theme.fontsizes.baseFontSize}
  }

  h1 {
    font-size: ${({ theme }) => theme.fontsizes.h1FontSize}
  }

  p, h2 {
    line-height: 1.7rem;
  }

  select {
    font-family: 'Roboto', sans-serif;
  }

  code {
    font-family: 'JetBrains Mono', Fallback, sans-serif !important;
  }

  /* PROGRESS BAR */
  #nprogress {
    pointer-events: none;
  }

  #nprogress .bar {
    background: rgb(231, 142, 165);
    background: linear-gradient(
      90deg,
      rgba(231, 142, 165, 1) 0%,
      rgba(254, 184, 206, 1) 74%
    );

    position: fixed;
    z-index: 1031;
    top: 0;
    left: 0;

    width: 100%;
    height: 2px;
  }
 /* SCROLL BAR */

  ::-webkit-scrollbar {
    width: 7px;
    height: 7px;
  }
  ::-webkit-scrollbar-button {
    width: 0px;
    height: 0px;
  }
  ::-webkit-scrollbar-thumb {
    background: #8a8a8a;
    border: 0px none #ffffff;
    border-radius: 11px;
  }
  ::-webkit-scrollbar-thumb:hover {
    background: #ffffff;
  }
  ::-webkit-scrollbar-thumb:active {
    background: #000000;
  }
  ::-webkit-scrollbar-track {
    background: ${({ theme }) => theme.baseBg};
    border: 0px none #ffffff;
    border-radius: 11px;
  }
  ::-webkit-scrollbar-track:hover {
    background: #666666;
  }
  ::-webkit-scrollbar-track:active {
    background: #333333;
  }
  ::-webkit-scrollbar-corner {
    background: transparent;
}


`;
