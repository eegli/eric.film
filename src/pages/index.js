import React from 'react';
import { request } from '../api/graphql';
import { IMGS_HOME } from '../api/queries';
import LayouContainer from '@/shared/layout/layout.container';
import {
  Sh1,
  Sh3,
  HeadingContainer,
  DescriptionContainer,
} from '@/shared/headings.styles';
import { Svideo } from '@/shared/elements.styles';

const IndexPage = ({ images }) => {
  return (
    <LayouContainer>
      <HeadingContainer>
        <Sh1>| eric.egli |</Sh1>
      </HeadingContainer>
      <DescriptionContainer>
        <Sh3>amateur filmmaker with a passion for frontend development</Sh3>
      </DescriptionContainer>
      {/* {images.map(img => (
        <img key={img.url} src={img.url} />
      ))} */}
      <Svideo autoPlay loop muted>
        <source src='/static/vid/home_teaser.mp4' type='video/mp4' />
      </Svideo>
    </LayouContainer>
  );
};
export async function getStaticProps() {
  const images = await request(IMGS_HOME);

  return {
    props: {
      images: images.imgCollectionsConnection.edges[0].node.collection,
    },
  };
}

export default IndexPage;
