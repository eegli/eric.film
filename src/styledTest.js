import { ThemeProvider } from 'styled-components';
import { mount, shallow } from 'enzyme';
import { theme } from '@/styles/theme';

const ThemeProviderWrapper = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export const shallowWithTheme = tree =>
  shallow(tree, {
    wrappingComponent: ThemeProviderWrapper,
  });

export const mountWithTheme = tree =>
  mount(tree, {
    wrappingComponent: ThemeProviderWrapper,
  });
