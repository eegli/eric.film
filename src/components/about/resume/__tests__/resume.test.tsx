import { mountWithTheme, shallowWithTheme } from '@/src/styledTest';
import Resume from '../resume.component';
import { ToggleContainer } from '../resume.styles';

describe('About - resume', () => {
  const mountWrapper = mountWithTheme(<Resume />);
  const shallowWrapper = shallowWithTheme(<Resume />);

  it('renders component', () => {
    expect(mountWrapper).toMatchSnapshot();
    // console.log(wrapper.debug());
  });

  it('has 4 categories', () => {
    expect(shallowWrapper.find('div')).toHaveLength(4);
  });

  it('has 2 containers that can be toggled', () => {
    expect(shallowWrapper.find(ToggleContainer)).toHaveLength(2);
  });
});
