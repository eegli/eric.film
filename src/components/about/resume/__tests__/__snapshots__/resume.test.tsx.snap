// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About - resume renders resume component 1`] = `
<DocumentFragment>
  .c0 {
  margin-top: 0.5rem;
  width: 93%;
  color: #121212;
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-right: 1rem;
}

.c1:nth-child(2) {
  margin-left: 1rem;
  margin-right: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #222222;
  color: #F1F1F1;
  padding: 1rem;
}

.c2:hover {
  cursor: pointer;
}

.c3 {
  display: inline;
  margin: 0;
  font-weight: 500;
}

@media screen and (max-width:1000px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media screen and (max-width:1000px) {
  .c1:nth-child(2) {
    margin: 2rem 0 0 0;
  }
}

@media screen and (max-width:1000px) {
  .c1 {
    margin: 0;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <p
          class="c3"
        >
          education - work
        </p>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 320 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
          />
        </svg>
      </div>
      <div
        class=""
        style="height: 0px; overflow: hidden; display: none; transition: height .25s ease;"
      />
    </div>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <p
          class="c3"
        >
          tools - technology
        </p>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 320 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
          />
        </svg>
      </div>
      <div
        class=""
        style="height: 0px; overflow: hidden; display: none; transition: height .25s ease;"
      />
    </div>
  </div>
</DocumentFragment>
`;
